### PR TITLE
Extraction of data from Jira

### PR DESCRIPTION
**Pull request by Kax-y**
Closes: #2 

The Jira issues endpoint can be reached at https://local.functions.nhost.run/v1/jira/. It requires email, API/PAT token, the domain name of the kanban board on Jira and the project key for the project you want the issues for. They should be sent as queries (i.e. ?projectKey=KAN&email=example@gmail.com&token=FaKeToKeN&name=exampledomain).

### CODE REVIEW

File: functions/jira/callAPI.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// import { Payload } from "./payload";
```
Comment (richyk1) [inline functions/jira/callAPI.ts:1]: Unnecessary comment

File: functions/jira/callAPI.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  let result = fetch(url, {
+    method: operation.method,
+    headers: headers,
+    body: body,
+  })
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      return response.json() as Promise<{ data: any }>;
```
Comment (richyk1) [inline functions/jira/callAPI.ts:39]: Is the data unknown? Maybe we could use the [unknown](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type) type here.

File: functions/jira/payload.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export type Payload = {};
```
Comment (richyk1) [inline functions/jira/payload.ts:1]: What is this?

**Review by richyk1 — APPROVED** [review]
Added some minor comments but in general it looks good! Too clean

File: functions/jira/index.ts
Comment (Felix-Thornblad) [inline functions/jira/index.ts:1]: did you test if 20-23 woks with edge cases or if .ToString is preferable? Am not sure myself but found
https://stackoverflow.com/questions/47664576/as-string-versus-tostring
where as string did not work and if it works like how it does in c# then according to 
https://stackoverflow.com/questions/2099900/difference-between-tostring-and-as-string-in-c-sharp
it would return null if a value is a number

**Review by Felix-Thornblad — APPROVED** [review]
Had a question about the as string used in index ts, but otherwise it looks good

File: functions/jira/index.ts
Comment (Dan-cmyk) [reply functions/jira/index.ts:1]: These are parsed from queries and can therefore be a string or undefined. We have already checked that they are not undefined which is why we assert them to string.
All types of invalid inputs will be caught by Jira which then would return an error.

**Review by gabrielfrimodig — APPROVED** [review]
